test/cqlpy/test_tablets.py

```diff
@@ -181,7 +181,38 @@ def test_describe_initial_tablets(cql, skip_without_tablets):
             "AND TABLETS = { 'initial' : 1 }"
     with new_test_keyspace(cql, ksdef) as keyspace:
         desc = cql.execute(f"DESCRIBE KEYSPACE {keyspace}")
-        assert "and tablets = {'initial': 1}" in desc.one().create_statement.lower()
+        assert "and tablets = {'enabled': true, 'initial': 1}" in desc.one().create_statement.lower()
+
+
+# Test to ensure metadata is printed when tablets are enabled during schema description, regardless of the use of WITH INTERNALS.
+#
+# Reproduces https://github.com/scylladb/scylladb/issues/22866
+@pytest.mark.parametrize("with_internals", [True, False])
+@pytest.mark.parametrize("tablets", [None, True, False])
+@pytest.mark.parametrize("initial", [None, 0, 7])
+def test_describe_tablets_internals(cql, with_internals, tablets, initial):
+    tablets_str = (f"'enabled': {tablets}" if tablets is not None else "").lower()
+    initial_str = f"'initial': {initial}" if initial is not None else ""
+
+    tablets_def = ""
+    if tablets_str or initial_str:
+        tablets_def = " AND TABLETS = {"
+        tablets_def += ", ".join(filter(None, [tablets_str, initial_str]))
+        tablets_def += "}"
+    ksdef = f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1' }}{tablets_def}"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        internals = " WITH INTERNALS" if with_internals else ""
+        desc = cql.execute(f"DESCRIBE SCHEMA{internals};")
+        results = desc.all()
+        validated = False
+        for res in results:
+            create_statement = res.create_statement.lower()
+            if f"create keyspace {keyspace}" in create_statement:
+                validated = True
+                assert tablets_str in create_statement
+                if tablets and initial is not None and initial > 0:
```

Comment:
For initial = None or 0 (as I noted above, they are the same...) it seems you expect the "initial" to be completely missing, right? You can assert that. Or, so be less stringent, you can asssert that either initial is missing **or** we have "initial: 0".